coderd/provisionerdserver/provisionerdserver.go

```diff
@@ -707,7 +708,9 @@ func (s *server) acquireProtoJob(ctx context.Context, job database.ProvisionerJo
 				PreviousParameterValues: convertRichParameterValues(lastWorkspaceBuildParameters),
 				VariableValues:          asVariableValues(templateVariables),
 				ExternalAuthProviders:   externalAuthProviders,
-				ReuseTerraformWorkspace: false, // TODO: Toggle based on experiment
+				// If active and experiment is enabled, allow workspace reuse existing TF
+				// workspaces (directories) for a faster startup.
+				ReuseTerraformWorkspace: activeVersion && s.Experiments.Enabled(codersdk.ExperimentTerraformWorkspace),
```

Comment:
There's a potential here for a TOCTOU race. `template` is set some time before this is called, which increases the likelihood.

This could lead to cache pollution, right?

Reply:
I don't think I totally follow. The `active` version is set before a workspace build (this only caches from workspace builds right now).

If the `active` version changes while a build is ongoing, the existing build will run as if it was the active. 

This is ok because the provisioners are single "threaded". The cache only keeps the latest active version directory. So if there is a race, the cache is fixed when the next template version comes in.

I guess in summary, the cache cannot be polluted because only 1 cached directory exists per provisioner. If the provisioner has an outdated view of the truth, it will be resolved in the next build. An eventually consistent approach.